app/models/broadcast.rb

```diff
@@ -10,6 +10,8 @@ class Broadcast < ApplicationRecord
   validates :banner_style, inclusion: { in: VALID_BANNER_STYLES }, allow_blank: true
   validate  :single_active_announcement_broadcast
 
+  before_save :update_last_active_at, if: :will_save_change_to_active?
```

Comment:
I was about to ask why this [came back](https://github.com/thepracticaldev/dev.to/pull/8451/files#r439868062), but now I see that `will_save_change_to` is [a Rails magic thing](https://apidock.com/rails/v6.0.0/ActiveRecord/AttributeMethods/Dirty/will_save_change_to_attribute%3F)!

Reply:
Correct! Although you were correct in that it was working properly initially.